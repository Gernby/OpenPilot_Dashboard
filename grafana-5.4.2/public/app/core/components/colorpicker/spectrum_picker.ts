/**
 * Wrapper for the new ngReact <color-picker> directive for backward compatibility.
 * Allows remaining <spectrum-picker> untouched in outdated plugins.
 * Technically, it's just a wrapper for react component with two-way data binding support.
 */
import coreModule from '../../core_module';

/** @ngInject */
export function spectrumPicker() {
  return {
    restrict: 'E',
    require: 'ngModel',
    scope: true,
    replace: true,
    template: '<color-picker color="ngModel.$viewValue" onChange="onColorChange"></color-picker>',
    link: (scope, element, attrs, ngModel) => {
      scope.ngModel = ngModel;
      scope.onColorChange = color => {
        ngModel.$setViewValue(color);
      };
    },
  };
}
coreModule.directive('spectrumPicker', spectrumPicker);
