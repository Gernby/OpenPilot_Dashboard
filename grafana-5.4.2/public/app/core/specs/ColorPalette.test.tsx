import React from 'react';
import renderer from 'react-test-renderer';
import { ColorPalette } from '../components/colorpicker/ColorPalette';

describe('CollorPalette', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ColorPalette color="#EAB839" onColorSelect={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
