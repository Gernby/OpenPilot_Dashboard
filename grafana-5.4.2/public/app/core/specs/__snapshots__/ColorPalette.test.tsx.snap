// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollorPalette renders correctly 1`] = `
<div
  className="graph-legend-popover"
>
  <p
    className="m-b-0"
  >
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#890f02",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#58140c",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#99440a",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#c15c17",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#967302",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#cca300",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#3f6833",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#2f575e",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#64b0c8",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#052b51",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#0a50a1",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#584477",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#3f2b5b",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#511749",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#e24d42",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#bf1b00",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#ef843c",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f4d598",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#e5ac0e",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#9ac48a",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#508642",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#6ed0e0",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#65c5db",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#0a437c",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#447ebc",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#614d93",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#d683ce",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#6d1f62",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#ea6460",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#e0752d",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f9934e",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#fceaca",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle-o"
      onClick={[Function]}
      style={
        Object {
          "color": "#eab839",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#b7dbab",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#629e51",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#70dbed",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#82b5d8",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#1f78c1",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#aea2e0",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#705da0",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#e5a8e2",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#962d82",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f29191",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#fce2de",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f9ba8f",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f9e2d2",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f2c96d",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#e0f9d7",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#7eb26d",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#cffaff",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#badff4",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#5195ce",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#dedaf7",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#806eb7",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#f9d9f9",
        }
      }
    >
       
    </i>
    <i
      className="pointer fa fa-circle"
      onClick={[Function]}
      style={
        Object {
          "color": "#ba43a9",
        }
      }
    >
       
    </i>
  </p>
</div>
`;
